### PR TITLE
adding required files for f35-py310 S2I image to built

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
## Related Issues and Dependencies

Related to https://github.com/thoth-station/s2i-thoth/issues/231

Includes linting changes for all `s2i_assemble.patch` files, which is outside this PR's scope, but otherwise `pre-commit` check won't pass.
Additionally added documentation to `README.rst` so people don't try to manually build the patch like I did...

## This introduces a breaking change

- [ ] Yes
- [X] No

## This Pull Request implements

Implements the files required to build a Thoth image of fedora35 with python3.10 using S2I.

## Description

Solely creates the base image, a solver image still needs to be built from this.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -9,6 +9,8 @@ s2i-thoth
 
 .. |s2i-thoth-f34-py39 on Quay| image:: https://quay.io/repository/thoth-station/s2i-thoth-f34-py39/status
 
+.. |s2i-thoth-f34-py39 on Quay| image:: https://quay.io/repository/thoth-station/s2i-thoth-f35-py310/status
```
Comment (goern) [inline README.rst:12]: this reads f34-py39, a c-n-p-probelm

File: ubi8-py36/s2i_assemble.patch [hunk 2]
```diff
@@ -1,5 +1,5 @@
---- a/assemble	2021-12-07 00:04:23.680648366 -0500
```
Comment (harshad16) [inline ubi8-py36/s2i_assemble.patch:1]: is changes to ubi file intentional ?

**Review by harshad16 — CHANGES_REQUESTED** [review]
nice work.
was the s2i_assemble.patch copied from other folders?

and another minor question

**Comment by Gregory-Pereira** [comment]
> nice work. was the s2i_assemble.patch copied from other folders?
> and another minor question
    > is changes to ubi file intentional ?

1. Yes I copied the file for the s2i_assemble.patch living in the `f34-py39` folder, and went to change anything specific to fedora version 34 or python version 3.9, but there were none. I did this because I wasn't exactly sure how these were meant to be implemented / generated, if there is a better way I am all ears.

2. I did not want to change the UBI assemble patches however, there was some formatting issue in them that made the pre-commit fail. I am guessing it was something in [this commit] that caused it, and I am not sure if changes were forced into master, the pre-commit was down, or the configuration file for precommit changed since then.

**Comment by harshad16** [comment]
> 1. Yes I copied the file for the s2i_assemble.patch living in the `f34-py39` folder, and went to change anything specific to fedora version 34 or python version 3.9, but there were none. I did this because I wasn't exactly sure how these were meant to be implemented / generated, if there is a better way I am all ears.

The reason why I asked this is because, the `s2i_assemble.patch` is created based on the difference of the `assemble`  present in the base image and addition we are trying to do from our assemble script. 
As the `assemble` bit can differ from base image to base image, it is good to verify this once.

*. Easiest way to verify: Try to build the image from the dockerfile. (if successful the patch is in good shape)

* How is the patch created (just for your knowledge, don't need to be done for this):
Get the `assemble` from the base image: usually located at `usr/libexec/s2i/assemble`
Remove the bits we don't want , (this can be referenced from other patches)

state the copied assemble script in `a/assemble`
state the removed bit assemble script in `b/assemble`
`diff -u a/assemble b/assemble > assemble.patch`

the resultant patch is what we require.

> 2. I did not want to change the UBI assemble patches however, there was some formatting issue in them that made the pre-commit fail. I am guessing it was something in [this commit](https://github.com/thoth-station/s2i-thoth/commit/60a0ea504dc82dd4063efb7b3358074152baecdb) that caused it, and I am not sure if changes were forced into master, the pre-commit was down, or the configuration file for precommit changed since then.

Thanks for the changes :)

**Comment by harshad16** [comment]
/hold

hold till the image build is confirmed.

**Comment by harshad16** [comment]
The s2i_assemble.patch script fails with f35 base image to be patched:
```
/usr/libexec/s2i ~
File assemble is read-only; trying to patch anyway
patching file assemble
Hunk #2 FAILED at 23.
1 out of 2 hunks FAILED -- saving rejects to file assemble.rej
```

**Comment by Gregory-Pereira** [comment]
Got it, thanks Harshad, taking a look now.

**Comment by Gregory-Pereira** [comment]
Can confirm that the `f35-py310` overlay now builds correctly:
```
$ podman build .
STEP 1/8: FROM registry.fedoraproject.org/f35/python3:0-41.container
...
STEP 8/8: USER 1001
--> Using cache 83dc730284d93fd5378c557bf4a3f1fc58647b0f8279eaab5aeb147de067fc78
--> 83dc730284d
83dc730284d93fd5378c557bf4a3f1fc58647b0f8279eaab5aeb147de067fc78
```

**Comment by Gregory-Pereira** [comment]
/hold
Working on formatting the `README.rst` file.

**Comment by Gregory-Pereira** [comment]
/unhold
Ready for review.

**Review by fridex — APPROVED** [review]
Changes look really nice to me. I've also verified the resulting container image works. I think this can go in. Thanks! 💯 👍🏻

**Pull request merged**